pagerduty/incident_workflow.go

```diff
@@ -31,16 +31,35 @@ type IncidentWorkflowStep struct {
 
 // IncidentWorkflowActionConfiguration represents the configuration for an incident workflow action
 type IncidentWorkflowActionConfiguration struct {
-	ActionID    string                         `json:"action_id,omitempty"`
-	Description *string                        `json:"description,omitempty"`
-	Inputs      []*IncidentWorkflowActionInput `json:"inputs,omitempty"`
+	ActionID          string                                    `json:"action_id,omitempty"`
+	Description       *string                                   `json:"description,omitempty"`
+	Inputs            []*IncidentWorkflowActionInput            `json:"inputs,omitempty"`
+	InlineStepsInputs []*IncidentWorkflowActionInlineStepsInput `json:"inline_steps_inputs,omitempty"`
 }
 
+// IncidentWorkflowActionInput represents the configuration for an incident workflow action input with a serialized string as the value
 type IncidentWorkflowActionInput struct {
 	Name  string `json:"name,omitempty"`
 	Value string `json:"value,omitempty"`
 }
 
+// IncidentWorkflowActionInlineStepsInput represents the configuration for an incident workflow action input with a series of inlined steps as the value
+type IncidentWorkflowActionInlineStepsInput struct {
+	Name  string `json:"name,omitempty"`
+	Value *IncidentWorkflowActionInlineStepsInputValue `json:"value,omitempty"`
+}
+
+// IncidentWorkflowActionInlineStepsInputValue represents the value for an inline_steps_input input
+type IncidentWorkflowActionInlineStepsInputValue struct {
+	Steps []*IncidentWorkflowActionInlineStep `json:"steps,omitempty"`
+}
+
+// IncidentWorkflowActionInlineStep represents a single step within an inline_steps_input input's value
+type IncidentWorkflowActionInlineStep struct {
+	Name          string                               `json:"name,omitempty"`
+	Configuration *IncidentWorkflowActionConfiguration `json:"action_configuration,omitempty"`
```

Comment:
I'm just curious to know if Incident Workflows API has input validation for this circular structure to avoid abuse of nested Inline Steps Configurations.

Reply:
Yes, we currently limit workflows to only 1 level. So if a step includes an action that has inline steps, a step of that inline workflow cannot itself use an action that has inline steps- the POST or PUT will 400 telling you it is disallowed.

That restriction may or may not be loosened in the future, but there will always be an upper limit for the number of "levels" enforced at the api boundary 👍

Reply:
Addendum, all of that validation is pushed to the service as much as possible. This change allows any level and relies on the server to respond with 400.

However, the terraform provider schema requires a set limit due to how it is constructed. In that case, I set the limit to `1` for now, but increasing it in the future should only need us to bump that integer to whatever number of levels we now allow.